mteb/filter_tasks.py

```diff
@@ -141,30 +142,45 @@ def _convert_to_set(domain: list[TaskDomain] | None) -> set:
 
     _tasks = []
     for t in tasks:
-        if langs_to_keep and not langs_to_keep.intersection(t.metadata.languages):
+        # Check if t is a class or an instance
+        is_class = isinstance(t, type)
+
+        # For metadata and superseded_by, we can access them directly
+        metadata = t.metadata
+        superseded_by = t.superseded_by
+
+        # Check if task is aggregate by checking class inheritance
+        if is_class:
+            # If it's a class, check if it's a subclass of AbsTaskAggregate
+            is_aggregate = issubclass(t, AbsTaskAggregate)
+        else:
+            # If it's an instance, check if it's an instance of AbsTaskAggregate
+            is_aggregate = isinstance(t, AbsTaskAggregate)
+
+        if langs_to_keep and not langs_to_keep.intersection(metadata.languages):
             continue
-        if script_to_keep and not script_to_keep.intersection(t.metadata.scripts):
+        if script_to_keep and not script_to_keep.intersection(metadata.scripts):
             continue
         if domains_to_keep and not domains_to_keep.intersection(
-            _convert_to_set(t.metadata.domains)
+            _convert_to_set(metadata.domains)
         ):
             continue
-        if task_types_to_keep and t.metadata.type not in task_types_to_keep:
+        if task_types_to_keep and metadata.type not in task_types_to_keep:
             continue
-        if categories_to_keep and t.metadata.category not in categories_to_keep:
+        if categories_to_keep and metadata.category not in categories_to_keep:
             continue
         if modalities_to_keep:
             if exclusive_modality_filter:
-                if set(t.metadata.modalities) != modalities_to_keep:
+                if set(metadata.modalities) != modalities_to_keep:
                     continue
             else:
-                if not modalities_to_keep.intersection(t.metadata.modalities):
+                if not modalities_to_keep.intersection(metadata.modalities):
                     continue
-        if exclude_superseded and t.superseded_by is not None:
+        if exclude_superseded and superseded_by is not None:
             continue
-        if exclude_aggregate and t.is_aggregate:
+        if exclude_aggregate and is_aggregate:
```

Comment:
I would revert everything else and simply do this one instead:
```suggestion
        is_aggregate = issubclass(t, AbsTaskAggregate) if isinstance(t, type) else isinstance(t, AbsTaskAggregate)
        if exclude_aggregate and is_aggregate:
```